tests/run_vector_tests.sh

```diff
@@ -9,18 +9,22 @@ fi
 
 pip install -r "$1/requirements.txt"
 $1/generator.py -t uptane --signature-encoding base64 -o vectors --cjson json-subset
-$1/server.py -t uptane --signature-encoding base64 &
+if [ "$2" == "valgrind" ]; then
+  $1/server.py -t uptane --signature-encoding base64 -P 8081 &
```

Comment:
httpclient_test.cc use bind(0) approach to get free port on the system. I suggest to create test_utils.h and move this implementation to this file. Then we could use this implementation in both tests.